### ISSUE TITLE
Issue with numbers on main clusters

### ISSUE THREAD

**Issue by mdominado** [nickcam/FlareClusterLayer#30]
Hello,

I'm currently using the 4.6 code of this library for a custom web application that uses the latest ArcGIS API. My main issue right now is for some reason the numbers for the main clusters show for a second or two and then disappear afterwards. Also, when I move the map, then stop moving the map, the numbers show but disappear as well.

I wanted to know if someone has any idea of what could be causing this?

I've attached a zip file that has a brief clip showing what I am running into.

[screen-capture.zip](https://github.com/nickcam/FlareClusterLayer/files/2094154/screen-capture.zip)

Any help would be appreciated!

**Comment by nickcam** [comment]
Hi mate, I'm not surprised it doesn't quite work using v4.7, there's usually breaking changes when ESRI updates the api. I haven't had a chance or a need to update it yet.

When I get around to it I was planning on doing a bit of a rewrite to make it fit the api object model a bit better (ie. less hacky stuff), and to preapre for webgl rendering of feature layers client side. Hopefully this will make it more resilient to api updates as well. When this will happen I can't say though. The only advice I have for the moment is to downgrade your project to use v4.6 of the js api if you really want to use the layer.

**Comment by mdominado** [comment]
Yeah, I definitely understand the feeling of making fixes whenever there are breaking changes from ESRI's API when they update it.

However, downgrading is not really an option for me. As I mentioned, I'm building a custom project for work, and coincidentally I already fixed whatever 4.6 broke in the application when I upgraded to the 4.7 API.

The company I work with is an ESRI gold partner, so I was wondering if you could explain what the API issue is, so that I can create a ticket for them to take a look at.

Thanks

**Comment by nickcam** [comment]
Hi man, yeah the updates that could have broken it wouldn't be ESRI's problem, 99% chance the bug would be in this layers code as it does some non-standard type things to draw...which is why I want to do a cleanup. Sending any issue to ESRI wouldn't get you much help.

I haven't looked at using 4.7 against it myself as I haven't had a chance to upgrade so I have no idea what the problems are yet. If it's only the numbers disappearing, you could probably just dig in and have a look at where the numbers are created and go from there, may even just be a css thing. Sorry I can't help more right now, just haven't had time to look at it. Hopefully sometime in the next couple of months though, maybe sooner.

**Comment by mdominado** [comment]
Hey Nick, just wanted to let you know that I fixed the issue with the text not staying properly on the clusters. The cause was due to a fix for an issue I had with the text symbols' haloSize and haloColor. The code was given to me by ESRI, however, it looks like the fix messed with all the SVGs on the map, whenever the map updates.

So, the issue wasn't with the flare cluster layer source code, but rather a workaround for the API that I had in place from ESRI.

Anyways, this issue should be resolved now.
